src/memory_prefetch.c

```diff
@@ -279,3 +279,30 @@ void removeClientFromPendingCommandsBatch(client *c) {
         }
     }
 }
+
+/* A simple way to prefetch the keys for a client command execution. This can be
+ * used for optimizing multi-key commands like mget, mset, etc. */
+void prefetchKeys(client *c, int first, int step, int count) {
+    /* Skip this for IO threads. Keys are already batch-prefetched. */
```

Comment:
Not all keys are prefetched in the context of io-threads. Up to 16 keys.

Reply:
That's right. What do you suggest? Move the check to the caller and skip prefetching the first 16 if io-thread are used?

Reply:
I wanted to optimize for ~10 keys. MGET with thousands of keys isn't a good pattern since it starves other clients, so that's less important IMO.

Reply:
Maybe MGET with large number of keys isn't a good pattern, however, clients still send large key batches in practice.

I suggest skipping the first 16 keys if IO threads is enabled and avoiding lookupKeyRead entirely, see my other comment.